src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/RazorLSPOptionsMonitor.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using MediatR;
+using Microsoft.Extensions.Options;
+using OmniSharp.Extensions.LanguageServer.Protocol.Server;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer
+{
+    internal class RazorLSPOptionsMonitor : IOptionsMonitor<RazorLSPOptions>
+    {
+        private readonly ILanguageServer _server;
+        private readonly RazorConfigurationService _configurationService;
+        private readonly IOptionsMonitorCache<RazorLSPOptions> _cache;
+        internal event Action<RazorLSPOptions, string> _onChange;
+        private RazorLSPOptions _currentValue;
+
+        public RazorLSPOptionsMonitor(ILanguageServer server, RazorConfigurationService configurationService, IOptionsMonitorCache<RazorLSPOptions> cache)
+        {
+            if (server is null)
+            {
+                throw new ArgumentNullException(nameof(server));
+            }
+
+            if (configurationService is null)
+            {
+                throw new ArgumentNullException(nameof(configurationService));
+            }
+
+            if (cache is null)
+            {
+                throw new ArgumentNullException(nameof(cache));
+            }
+
+            _server = server;
+            _configurationService = configurationService;
+            _cache = cache;
+            Initialize();
+        }
+
+        public RazorLSPOptions CurrentValue => Get(Options.DefaultName);
+
+        public RazorLSPOptions Get(string name)
+        {
+            name = name ?? Options.DefaultName;
+            return _cache.GetOrAdd(name, () => _currentValue);
+        }
+
+        public IDisposable OnChange(Action<RazorLSPOptions, string> listener)
+        {
+            var disposable = new ChangeTrackerDisposable(this, listener);
+            _onChange += disposable.OnChange;
+            return disposable;
```

Comment:
Does this ever get disposed? At what point in the life-cycle would that be a good idea?

Reply:
If someone registers a callback with `OnChange`, they will be returned this `IDisposable`. It is up to them to dispose it when they no longer want to be called when something changes.